### ISSUE TITLE
Notfies :before property doubles the leaf-directory

### ISSUE THREAD

**Issue by TomKemperNL** [sous-chefs/ark#196]
### Cookbook version
3.1.0

### Chef-client version
12.19.36

### Platform Details
Windows 10 (15063)

### Scenario:
Combining the ark resource :put action with notifies

### Steps to Reproduce:
My resource is declared like 
```
ark 'some_application' do
  action :put
  name "app_test"
  url "someurl"
  path "C:/Temp"
  strip_components 0
  notifies :stop, 'service[someService]', :before
end
```

### Expected Result:
I would expect the zip to be expanded to C:/Temp/app_test

### Actual Result:
The zip is expanded to C:/Temp/app_test/app_test.
If I comment out the notifies property, it works correctly.
If I only use notified :immediate or :delayed, it works correctly

**Comment by TomKemperNL** [comment]
Cannot reproduce after upgrading to Chef-client 13.2.20.

**Comment by TomKemperNL** [comment]
Nope, it wasn't gone. I just retested with Chef Client 13.2.20 while it was still commented out.

Furthermore a colleague of mine has the same problem on Ubuntu 16.

**Comment by TomKemperNL** [comment]
This is probably caused by the fact that notify :before triggers a why-run. However action :put will call set_put_paths during both the real and the why-run, joining the resource name onto the path twice.

**Comment by TomKemperNL** [comment]
I'm somewhat concerned by the complete silence. Is there something wrong with the issue? Is there anything I can do to help? Is everyone just on summer vacation?

**Comment by dosyfier** [comment]
I just faced the very same issue on one of my projects, corrected it and saw this 2-year old ticket.
I submitted a PR with modifications to support the why-run mode and correct the problem described here. Hopefully it will be accepted...

**Comment by damacus** [comment]
I need to check if this is still a bug, but I highly suspect it is!
